pkg/controller/jenkins/configuration/base/resources/deployment.go

```diff
@@ -0,0 +1,59 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+)
+
+// NewJenkinsMasterPod builds Jenkins Master Kubernetes Pod resource.
+func NewJenkinsDeployment(objectMeta metav1.ObjectMeta, jenkins *v1alpha2.Jenkins) *appsv1.Deployment {
+	serviceAccountName := objectMeta.Name
+	objectMeta.Annotations = jenkins.Spec.Master.Annotations
+	objectMeta.Name = GetJenkinsDeploymentName(jenkins)
+	selector := &metav1.LabelSelector{MatchLabels: objectMeta.Labels}
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      objectMeta.Name,
+			Namespace: objectMeta.Namespace,
+			Labels:    objectMeta.Labels,
+			OwnerReferences: []metav1.OwnerReference{
```

Comment:
remove `OwnerReference` as `err := r.CreateResource(jenkinsDeployment)` will set it for you